app/imports/api/ethers/ethersFunctions.js

```diff
@@ -26,101 +27,94 @@ export async function deployContracts(contract) {
+
   // load provider
   const provider = loadProvider();
   //  tenent is new signer of next contract call
   const signer = provider.getSigner(tenant.address);
   // create new contract instance
   const contractInstance = new ethers.Contract(contract.address, contract.abi, signer);
+  saveTransactionForRecord(contract.transactionLog, currentD, tenantToContract);
   // call contract to pay rent
   contractInstance.payRent(contract.homeowner.address, { value: rentEth });
+  saveTransactionForRecord(contract.transactionLog, currentD, contractToHomeowner);
 }
 
 export async function destroyContract(contract) {
   const contractInstance = new ethers.Contract(contract.address, contract.abi, contract.homeowner.signer);
   contractInstance.close();
 }
 
-function determineNextPayment(tPayPeriod, currentD) {
-  let nextPaymentDate = null;
-  // update next pay period
-  switch (tPayPeriod) {
-  case 'months':
-    nextPaymentDate = moment(currentD).add('1', 'month');
-    break;
-  case 'weeks':
-    nextPaymentDate = moment(currentD).add('1', 'week');
-    break;
-  case 'seconds':
-    nextPaymentDate = moment(currentD).add('1', 'seconds');
-    break;
-  default:
-    console.log(`${tPayPeriod} is not a proper response`);
-  }
-  return nextPaymentDate;
-}
-
-export async function payRentFunction(contract) {
+export async function payRentScheduler(contract) {
   // Initialize variables
   const dateDeployed = moment();
   let nextPaymentDate = null;
   let currentDate = dateDeployed;
+  const tenant = contract.tenants[0];
+  const tenantPayPeriod = tenant.period;
+  let finalPaymentDate = null;
+  let counter = 1;
+  const destroyString = `Contract at ${contract.address} destroyed`;
 
-  const tenantPayPeriod = contract.tenants[0].period;
-  const finalPaymentDate = moment(dateDeployed).add('1', 'year');
+  finalPaymentDate = moment(dateDeployed).add('1', 'year');
+
+  // used for debugging
+  if (tenantPayPeriod === 'seconds') {
+    finalPaymentDate = moment(dateDeployed).add('10', 'seconds');
+  }
 
   // interval function
   const paymentInterval = function () {
+
     currentDate = nextPaymentDate;
     if (moment(currentDate).isBefore(moment(finalPaymentDate))) {
       nextPaymentDate = determineNextPayment(tenantPayPeriod, currentDate);
+      counter++;
       // pay the rent
+      console.log(`Transaction ${counter}`);
       payRent(contract);
       setTimeout(paymentInterval, moment(nextPaymentDate).diff(currentDate));
     } else {
       // final payment, destroy the contract
-      payRent(contract);
+      saveTransactionForRecord(contract.transactionLog, moment(currentDate).format(), destroyString);
       destroyContract(contract);
     }
   };
 
   // run the recursive timeout
+  payRent(contract, currentDate);
+  console.log(`Transaction ${counter}`);
   nextPaymentDate = determineNextPayment(tenantPayPeriod, currentDate);
   setTimeout(paymentInterval, moment(nextPaymentDate).diff(moment(currentDate)));
```

Comment:
you might want to check if nextPaymentDate === null before setting the timeout or work with a default timeout if it is null. or you update determineNextPayment to return some valid default instead of a null default.

Reply:
done